### PR TITLE
feat: migrate code from googleapis/java-mediatranslation

### PR DESCRIPTION
**Pull request by anguillanneuf**
Towards internal [b/255375738](https://b.corp.google.com/issues/255375738):

- feat: initial generation
- samples: use the released artifact directly (#11)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.13 (#19)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.14 (#23)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.15 (#25)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.1.1 (#31)
- media-translation-sample: translate from file (#15)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.16 (#41)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.17 (#45)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.1.2 (#49)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.18 (#77)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.0 (#89)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.2 (#103)
- test(deps): update dependency junit:junit to v4.13.1 (#138)
- test(deps): update dependency com.google.truth:truth to v1.1 (#147)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.21 (#133)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.4 (#150)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.6 (#164)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.7 (#179)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.9 (#192)
- test(deps): update dependency com.google.truth:truth to v1.1.2 (#204)
- test(deps): update dependency junit:junit to v4.13.2 (#220)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.11 (#215)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.2.12 (#234)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.3.0 (#246)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.22 (#253)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.3.1 (#260)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.3.2 (#273)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.4.0 (#286)
- test(deps): update dependency com.google.truth:truth to v1.1.3 (#299)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.5.0 (#298)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.0.23 (#307)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.5.1 (#306)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.5.2 (#320)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.6.0 (#509)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.6.1 (#516)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.6.2 (#526)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.0 (#528)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.1 (#540)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.2 (#546)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.3 (#558)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.4 (#577)
- chore(deps): update dependency com.google.cloud.samples:shared-configuration to v1.2.0 (#574)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.5 (#587)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.9 (#628)
- docs(samples): remove unsupported field RecognitionResult (#631)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.10 (#636)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.7.11 (#641)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.8.0 (#654)
- feat: Enable REST transport for most of Java and Go clients (#658)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.8.1 (#662)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.9.0 (#678)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.9.2 (#682)
- chore(bazel): Update WORKSPACE files for rules_gapic, gax_java, generator_java versions (#685)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.9.3 (#689)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.9.4 (#694)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.9.5 (#724)
- chore(deps): update dependency com.google.cloud:google-cloud-mediatranslation to v0.9.6 (#728)

Fixes #issue

> It's a good idea to open an issue first for discussion.

- [x] I have followed [Sample Format 
Guide](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md)
- [x] `pom.xml` parent set to latest `shared-configuration`
- [x] Appropriate changes to README are included in PR
- [x] API's need to be enabled to test (tell us)
- [x] Environment Variables need to be set (ask us to set them)
- [x] **Tests** pass:   `mvn clean verify` **required**
- [x] **Lint**  passes: `mvn -P lint checkstyle:check` **required**
- [x] **Static Analysis**:  `mvn -P lint clean compile pmd:cpd-check spotbugs:check` **advisory only**
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

File: mediatranslation/generated/com/google/cloud/mediatranslation/v1beta1/speechtranslationservicesettings/streamingtranslatespeech/SyncStreamingTranslateSpeech.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2022 Google LLC
```
Comment (averikitsch) [inline mediatranslation/generated/com/google/cloud/mediatranslation/v1beta1/speechtranslationservicesettings/streamingtranslatespeech/SyncStreamingTranslateSpeech.java:2]: All files under generated should not be migrated

Comment (anguillanneuf) [reply mediatranslation/generated/com/google/cloud/mediatranslation/v1beta1/speechtranslationservicesettings/streamingtranslatespeech/SyncStreamingTranslateSpeech.java]: Oh good point. They have been taken out.

File: mediatranslation/README.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
+[//]: # "This README.md file is auto-generated, all changes to this file will be lost."
```
Comment (averikitsch) [inline mediatranslation/README.md:1]: We need to decide if we want to use this readme and if we want to create sub-directories within the product directory to prepare for future (separate) samples

Comment (anguillanneuf) [reply mediatranslation/README.md:1]: Ack.

**Comment by averikitsch** [comment]
2 open questions ([go/java-monorepo-migration-faq](http://goto.google.com/java-monorepo-migration-faq)), should we add a snippets directory? and should we delete the generated readme? I think yes, do you agree?

**Comment by anguillanneuf** [comment]
@averikitsch Is this PR ready for merging now?

**Review by averikitsch — APPROVED** [review]

**Pull request merged**